kyuubi-common/src/main/scala/org/apache/kyuubi/service/ThriftFrontendService.scala

```diff
@@ -105,11 +105,11 @@ class ThriftFrontendService private(name: String, be: BackendService, oomHook: R
     super.initialize(conf)
   }
 
-  override def connectionUrl(server: Boolean = false): String = {
+  override def connectionUrl: String = {
     getServiceState match {
       case s @ ServiceState.LATENT => throw new IllegalStateException(s"Illegal Service State: $s")
       case _ =>
-        if (server || conf.get(ENGINE_CONNECTION_URL_USE_HOSTNAME)) {
+        if (conf.get(ENGINE_CONNECTION_URL_USE_HOSTNAME)) {
```

Comment:
this is an engine config, we need to create `SparkSQLFrontendService` to remove  the `server` flag

Reply:
Did you mean this PR introduce the side-effort? When diving into the code base, I found the server flag only be used at [here](https://github.com/apache/incubator-kyuubi/blob/master/kyuubi-common/src/main/scala/org/apache/kyuubi/service/ThriftFrontendService.scala#L131), right? Before this PR, the same logic is also been judged in `KyuubiServer`(KyuubiServer used ThriftFrontendService).

In fact, I haven't thought about the real role of SparkSQLFrontendService. Will it be a CompositeService? ThriftFrontendService will still be used by KyuubiServer. You mean we need to put the judgment logic of the engine configuration in CompositeService (SparkSQLFrontendService)?